dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala

```diff
@@ -330,6 +330,73 @@ case class BinaryOutcomeDLCClient(
     }
   }
 
+  def createUnsignedMutualClosePSBT(
+      sig: SchnorrDigitalSignature,
+      fundingTx: Transaction): PSBT = {
+    val (toLocalPayout, toRemotePayout) =
+      if (Schnorr.verify(messageWin, sig, oraclePubKey)) {
+        (winPayout, remoteWinPayout)
+      } else if (Schnorr.verify(messageLose, sig, oraclePubKey)) {
+        (losePayout, remoteLosePayout)
+      } else {
+        throw new IllegalArgumentException(
+          s"Signature does not correspond to either possible outcome! $sig")
+      }
+
+    val toLocal =
+      TransactionOutput(toLocalPayout,
+                        P2WPKHWitnessSPKV0(finalPrivKey.publicKey))
+    val toRemote =
+      TransactionOutput(toRemotePayout, P2WPKHWitnessSPKV0(finalRemotePubKey))
+    val outputs = if (isInitiator) {
+      Vector(toLocal, toRemote)
+    } else {
+      Vector(toRemote, toLocal)
+    }
+    val input = TransactionInput(
+      TransactionOutPoint(fundingTx.txId, UInt32.zero),
+      EmptyScriptSignature,
+      sequence)
+    val utx = BaseTransaction(TransactionConstants.validLockVersion,
+                              Vector(input),
+                              outputs,
+                              UInt32.zero)
+
+    PSBT
+      .fromUnsignedTx(utx)
+      .addUTXOToInput(fundingTx, index = 0)
+      .addScriptWitnessToInput(P2WSHWitnessV0(fundingSPK), index = 0)
+  }
+
+  def createMutualCloseTxSig(
+      sig: SchnorrDigitalSignature,
+      fundingTx: Transaction): Future[PartialSignature] = {
+    val unsignedPSBT = createUnsignedMutualClosePSBT(sig, fundingTx)
+
+    unsignedPSBT
+      .sign(inputIndex = 0, fundingPrivKey)
+      .map(
+        _.inputMaps.head.partialSignatures
+          .find(_.pubKey == fundingPrivKey.publicKey)
+          .get)
```

Comment:
rather than using `.get here it would be nice to just pattern match and then give a better error message in the `case None => ` case. 

I.e. 
```scala
case None => sys.error(s"You have constructed the psbt incorrectly! We could not find a public key=${fundingPrivKey.publicKey}")
```